### PR TITLE
Reader: add conversations list component

### PR DESCRIPTION
**Pull request by bluefuton**
For the upcoming Conversations tool. Accepts a blog ID, post ID and a list of comment IDs, then outputs a flat list of comments, like so:

screen shot 2017-07-19 at 16 53 38

### To test

Take a look at Devdocs:

http://calypso.localhost:3000/devdocs/blocks/conversation-comment-list

### CODE REVIEW

File: client/blocks/conversations/list.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	render() {
+		const { commentIds, commentsTree } = this.props;
+		if ( ! commentIds ) {
+			return null;
+		}
+
+		return (
+			<div className="conversations__comment-list">
+				{ map( commentIds, commentId =>
+					<PostComment
```
Comment (blowery) [inline client/blocks/conversations/list.jsx]: it's not doc'd but this needs the `post` too. we'll have to hand in it..

**Comment by bluefuton** [comment]
Currently looks like this:

screen shot 2017-07-21 at 19 21 08

Todo for future PRs:
- hook up reply button
- formatting (date to move to first line, content to move up)

File: client/blocks/comments/post-comment.jsx [hunk 2]
```diff
@@ -53,7 +53,7 @@ class PostComment extends Component {
 	renderRepliesList() {
 		const commentChildrenIds = get( this.props.commentsTree, [ this.props.commentId, 'children' ] );
 		// Hide children if more than maxChildrenToShow, but not if replying
-		const exceedsMaxChildrenToShow = commentChildrenIds.length < this.props.maxChildrenToShow;
+		const exceedsMaxChildrenToShow = commentChildrenIds && commentChildrenIds.length < this.props.maxChildrenToShow;
```
Comment (bluefuton) [inline client/blocks/comments/post-comment.jsx]: Thanks for spotting this @samouri!

File: client/blocks/conversations/list.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+			return null;
+		}
+
+		return (
+			<ul className="conversations__comment-list">
+				{ map( commentIds, commentId => {
+					return (
+						<PostComment
+							commentsTree={ commentsTree }
+							key={ commentId }
```
Comment (samouri) [inline client/blocks/conversations/list.jsx]: I feel like this shouldn't need the tree.  it should be connected and grab the tree itself.

**Comment by samouri** [comment]
> then outputs a flat list of comments, like so:

Do we only want to show top level comments? Or do we want to flatten all of the comments?

We can iterate on the behavior but this is going to be interesting :)

**Comment by blowery** [comment]
> Do we only want to show top level comments? Or do we want to flatten all of the comments?

it's just the last three comments, flat. They could be anywhere in the tree.

File: client/blocks/conversations/list.jsx (hunk 3 above)
Comment (bluefuton) [inline client/blocks/conversations/list.jsx]: Yep - there's actually a note to that effect in `PostComment` from the original developer I believe :)

**Pull request merged**